ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -184,271 +124,257 @@ class GameManager {
-                             correct: false,
-                             start_timestamp: 0, // вже намальовано
-                             tag: Constants.tag,
-                             trial_index: countTest + 1,
-                             start_time: startTrialTimestamp * 1000,
-                             response_touch_timestamp: 0)
+    guard let startTrialTimestamp = startTrialTimestamp else { return }
+
+    let model = FlankerModel(
+        rt: 0.0,
+        stimulus: text,
+        button_pressed: nil,
+        image_time: endTrialTimestamp! * 1000, // має намалювати
+        correct: false,
+        start_timestamp: 0, // вже намальовано
+        tag: Constants.tag,
+        trial_index: countTest + 1,
+        start_time: startTrialTimestamp * 1000,
+        response_touch_timestamp: 0
+    )
 
     resultManager.addStepData(data: model)
-    delegate?.resultTest(avrgTime: nil, procentCorrect: nil, data: model, dataArray: nil, isShowResults: gameParameters.showResults, minAccuracy: gameParameters.minimumAccuracy)
-    responseText = Constants.timeRespondText
+    delegate?.resultTest(avrgTime: nil, procentCorrect: nil, data: model, dataArray: nil, isShowResults: false, minAccuracy: gameParameters.minimumAccuracy)
 
-    let timer = Timer(timeInterval: Constants.lowTimeInterval, target: self, selector: #selector(self.setDefaultText), userInfo: nil, repeats: false)
-    RunLoop.main.add(timer, forMode: .common)
+    if gameParameters.showFeedback {
+      let timer = Timer(timeInterval: Constants.lowTimeInterval, target: self, selector: #selector(setDefaultText), userInfo: nil, repeats: false)
+      RunLoop.main.add(timer, forMode: .common)
+    } else {
+      setDefaultText(isFirst: false)
+    }
   }
 
-  func clearData() {
-    resultManager.cleanData()
-    countTest = 0
-    correctAnswers = 0
-    arrayTimes = []
-    invalidateTimers()
-  }
-}
+  func handleEndOfGame() {
+    guard let gameParameters = gameParameters else { return }
 
-private extension GameManager {
-  func randomString(length: Int = 5) -> String {
-    let letters = "<>"
-    let lettersArray = Array(letters)
-    let symbolArray = [">>", "<<", "--"]
+    endFeedbackTimestamp = CACurrentMediaTime()
+    setEndTimeViewingImage(time: endFeedbackTimestamp!, isStart: false, type: .feedback)
 
-    let randSymbol = arc4random_uniform(UInt32(symbolArray.count))
-    let randLetters = arc4random_uniform(UInt32(lettersArray.count))
+    let sumArray = arrayTimes.reduce(0, +)
+    let avrgArray = arrayTimes.count > 0 ? sumArray / arrayTimes.count : 0
+    let procentsCorrect = Float(correctAnswers) / Float(countAllGame) * 100
 
-    return symbolArray[Int(randSymbol)] + String(lettersArray[Int(randLetters)]) + symbolArray[Int(randSymbol)]
+    clearData()
+
+    delegate?.updateText(text: "Game Over", color: .black, font: Constants.bigFont, isStart: false, typeTime: .feedback)
```

Comment:
What is the purpose of this game over screen?

Reply:
Initially it was a screen for I check the end of the cycle, but I Kept the screen, I can remove  it.